src/operator/contrib/transformer-inl.h

```diff
@@ -41,7 +41,9 @@ static void DivSqrtDimForward_(const nnvm::NodeAttrs& attrs,
                   const std::vector<OpReqType>& req,
                   const std::vector<TBlob>& outputs) {
   mshadow::Stream<xpu> *s = ctx.get_stream<xpu>();
-  double sqrt_dim = std::sqrt(static_cast<double>(inputs[0].shape_[inputs[0].ndim() - 1]));
+  int input_ndim = inputs[0].ndim();
+  int last_idx = (input_ndim == 0) ? (0) : (input_ndim - 1);
+  double sqrt_dim = std::sqrt(static_cast<double>(inputs[0].shape_[last_idx]));
```

Comment:
If `input_ndim = 0`, `inputs[0]` is a scalar, and `inputs[0].shape_` will have nothing. I think instead of making these changes, we should just add one check: `CHECK_GT(inputs[0].ndim(), 1)`?

Reply:
Good catch! I missed it.